### PR TITLE
Restore Win32-2.6.2.0 workaround

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: ouroboros-consensus/src-win32/Ouroboros/Consensus/Storage/Seek.hsc
Comment (newhoggy) [inline ouroboros-consensus/src-win32/Ouroboros/Consensus/Storage/Seek.hsc:27]: This is function is re-introduced because it was introduced in `Win32-2.6.2.0` which we can't use yet.

**Review by Jasagredo — APPROVED** [review]
Looks good to me. The note on the cabal file and the constraints on the GHC version look super useful. I would still prefer to get the approval from @nfrisby and/or @coot 

Thanks!

**Comment by nfrisby** [comment]
I defer to @coot. Thanks.

File: ouroboros-consensus/ouroboros-consensus.cabal
Comment (coot) [inline ouroboros-consensus/ouroboros-consensus.cabal:329]:
```suggestion
    if impl(ghc >= 8.10.2) && impl(ghc < 8.11)
```
There is no `ghc-8.11` isn't it?

File: ouroboros-consensus/src-win32/Ouroboros/Consensus/Storage/Seek.hsc
Comment (coot) [inline ouroboros-consensus/src-win32/Ouroboros/Consensus/Storage/Seek.hsc:25]: I don't think we support `32` bit architectures at all, but it's ok.

**Review by coot — CHANGES_REQUESTED** [review]

File: ouroboros-consensus/ouroboros-consensus.cabal
Comment (newhoggy) [reply ouroboros-consensus/ouroboros-consensus.cabal]: I've moved it to `< ghc-8.11`, although I've never seen a `ghc-8.11` in the wild. 😁

**Review by coot — APPROVED** [review]
LGTM

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**
